### PR TITLE
DAOS-10674 build: Add Ubuntu to build

### PR DESCRIPTION
**Pull request by jolivier23**
Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Comment by jolivier23** [comment]
@brianjmurrell I may need some help figuring out what to change on this.

File: ucx.spec [hunk 1]
```diff
@@ -105,6 +105,7 @@ Provides header files and examples for developing with UCX.
            --disable-assertions \
            --disable-params-check \
            --without-java \
+           --prefix=%{_prefix} \
```
Comment (brianjmurrell) [inline ucx.spec]: The `%configure` macro will include this itself.

```suggestion
```

File: ucx.spec [hunk 2]
```diff
@@ -121,8 +122,8 @@ Provides header files and examples for developing with UCX.
 make %{?_smp_mflags} V=1
 
 %install
-make DESTDIR=%{buildroot} install
-rm -f %{buildroot}%{_libdir}/*.la
+make DESTDIR=%{buildroot} install prefix=%{_prefix} libdir=%{_libdir} datadir=%{_datadir}
```
Comment (brianjmurrell) [inline ucx.spec]: These shouldn't be necessary.  IIUC the RPM builds were already working correctly so probably nothing in this file needs updating.

```suggestion
make DESTDIR=%{buildroot} install
```

File: ucx.spec [hunk 3]
```diff
@@ -121,8 +122,8 @@ Provides header files and examples for developing with UCX.
...
+rm -f %{buildroot}%{_libdir}/*la
```
Comment (brianjmurrell) [inline ucx.spec]: It really is only `.la` files we want to remove though, right?

```suggestion
rm -f %{buildroot}%{_libdir}/*.la
```

File: ucx.spec [hunk 4]
```diff
@@ -313,6 +314,9 @@ library internals, protocol objects, transports status, and more.
 %endif
 
 %changelog
+* Wed Jun 01 2022 Jeff Olivier <jeffrey.v.olivier@intel.com> - 1.12.1-2
+- Moved source to make file, add prefix
+
```
Comment (brianjmurrell) [inline ucx.spec]: I don't think anything in the RPM build needs updating.

```suggestion
```

File: debian/patches/series [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# static-bfd.patch
+undo-upstream.patch
```
Comment (brianjmurrell) [inline debian/patches/series]: I wonder if this patch really does get applied.  Our debian builds actually take their patches from the specfile.  I'm not sure if we will apply patches from this dir if they exist.  I will check the build.

So, yeah, this is the problem.  If you add this patch to the RPM specfile, things should work better.

I suppose we really ought to handle this situation better.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by brianjmurrell — DISMISSED** [review]

File: debian/control [hunk 6]
```diff
@@ -42,7 +42,6 @@ Description: Header files for UCX library
 Package: libucx0
 Section: libs
 Depends: ${shlibs:Depends}, ${misc:Depends}
-Breaks: libopenmpi3 ( << 4.1.0-3)
```
Comment (brianjmurrell) [inline debian/control]: I'd rather see this commented out with a comment that **we** did that, not upstream, and why.  Just that in the future, we know.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**